code/utils/threading.cpp

```diff
@@ -85,6 +87,23 @@ namespace threading {
 			return num_cores;
 		}
 	}
+#elif defined __APPLE__
+	static size_t get_number_of_physical_cores() {
+		int num;
+		size_t numSize = sizeof(num);
+
+		// apple silicon (performance cores only)
+		if ( !sysctlbyname("hw.perflevel0.physicalcpu", &num, &numSize, nullptr, 0) ) {
+			return num;
```

Comment:
are we guaranteed that these return 1 or larger?
If this can somehow fail or return 0, we need to catch it, as downstream code assumes that the minimum number of cores detected is 1